torch/distributed/distributed_c10d.py

```diff
@@ -1465,6 +1466,62 @@ def reduce_scatter(output,
         work.wait()
 
 
+def alltoall(output,
+             input,
+             group=group.WORLD,
+             async_op=False):
+    """
+    Each process splits input tensor equally and then scatters the split list to all processes in a group.
+    Then concatenate the received tensors from all the processes in the group and return single output tensor.
+
+    Arguments:
+        output (Tensor): Gathered cancatenated output tensor.
+        input (Tensor): Input tensor to scatter.
+        group (ProcessGroup, optional): The process group to work on.
+        async_op (bool, optional): Whether this op should be an async op.
+
+    Returns:
+        Async work handle, if async_op is set to True.
+        None, if not async_op or if not part of the group.
+
+    Example:
+        input:
+            [ 0  1  2  3]  # Rank 0
+            [ 4  5  6  7]  # Rank 1
+            [ 8  9 10 11]  # Rank 2
+            [12 13 14 15]  # Rank 3
+
+        output:
+            [ 0  4  8 12]  # Rank 0
+            [ 1  5  9 13]  # Rank 1
+            [ 2  6 10 14]  # Rank 2
+            [ 3  7 11 15]  # Rank 3
+
+        Essentially, it is similar to following operation:
+            scatter_list = list(input.chunk(world_size))
```

Comment:
Will this show as a code block in the doc. We might need to add `>>>` prefix here, e.g.:

https://github.com/pytorch/pytorch/blob/02f09a1bbd3838a8384c1ea69bf314efb60b0439/torch/distributed/rpc/api.py#L181-L204


Could you please build the doc locally and post a screenshot of the API doc?

You can server the page using [this script](https://gist.github.com/colesbury/df218a96334dbd31ee0e93760f70f114) locally.

Reply:
I will update the example and doc as we finalize on API semantics and arguments. Will add `>>>` prefix at that time.